java-checks/src/main/java/org/sonar/java/checks/SwitchLastCaseIsDefaultCheck.java

```diff
@@ -57,7 +58,14 @@ private static boolean getDefaultLabel(SwitchStatementTree switchStatementTree)
   }
 
   private static boolean isDefault(CaseLabelTree caseLabelTree) {
-    return JavaKeyword.DEFAULT.getValue().equals(caseLabelTree.caseOrDefaultKeyword().text());
+    if (equalsDefaultKeyword(caseLabelTree.caseOrDefaultKeyword().text())) {
+      return true;
+    }
+    return caseLabelTree.expressions().stream().anyMatch(expr -> equalsDefaultKeyword(expr.firstToken().text()));
```

Comment:
To be honest, I preferred the comparison with the `DEFAULT_PATTERN` tree kind.
It is indeed deprecated, but only because this is a preview feature of the JDK not yet officialized. In the meantime, we still want to be able to parse this code, so it's valid. We will remove the deprecation as soon as the feature is final, and so the issue will disappear. I'm OK living with the issue.